ancv/web/server.py

```diff
@@ -258,3 +262,174 @@ def server_timing_header(timings: dict[str, timedelta]) -> str:
         f"{name.replace(' ', '-')};dur={duration // timedelta(milliseconds=1)}"
         for name, duration in timings.items()
     )
+
+
+class WebHandler(Runnable):
+    """A handler serving a rendered template loaded from a URL with periodic refresh."""
+
+    def __init__(
+        self, destination: str, refresh_interval: timedelta = timedelta(seconds=300)
+    ) -> None:
+        """Initializes the handler.
+
+        Args:
+            destination: The URL to load the JSON Resume from.
+            refresh_interval: How often to refresh the resume.
+        """
+        self.destination = destination
+        self.refresh_interval = refresh_interval
+        self.cache: str = ""
+        self.last_fetch: float = 0
+        self._last_valid_render: str = ""
+
+        LOGGER.debug("Instantiating web application.")
+        self.app = web.Application()
+
+        LOGGER.debug("Adding routes.")
+        self.app.add_routes([web.get("/", self.root)])
+
+        self.app.cleanup_ctx.append(self.app_context)
+
+    def run(self, context: ServerContext) -> None:
+        LOGGER.info("Loaded, starting server...")
+        web.run_app(self.app, host=context.host, port=context.port, path=context.path)
+
+    async def app_context(self, app: web.Application) -> AsyncGenerator[None, None]:
+        """Sets up the application context with required clients.
+
+        Args:
+            app: The app instance to attach our state to.
+        """
+        log = LOGGER.bind(app=app)
+        log.debug("App context initialization starting.")
+
+        log.debug("Starting client session.")
+        session = ClientSession()
+        app["client_session"] = session
+        log.debug("Started client session.")
+
+        log.debug("App context initialization done, yielding.")
+        yield
+
+        log.debug("App context teardown starting.")
+        await session.close()
+        log.debug("App context teardown done.")
+
+    async def fetch(self, session: ClientSession) -> ResumeSchema | web.Response:
+        """Fetches and validates resume JSON from the destination URL.
+
+        Args:
+            session: The aiohttp client session to use for requests.
+
+        Returns:
+            ResumeSchema: The validated resume data
+
+        Raises:
+            ResumeLookupError: When resume cannot be fetched from destination
+            json.JSONDecodeError: When response is not valid JSON
```

Comment:
This one (and I think others) isn't actually thrown, is it? It's swallowed, and callers won't need to/can't handle it

Reply:
Docstrings updated, please let me know if you notice something wrong. That was my bad, wrote in the beginning, and after few changes, they got outdated/wrong, I did reviewed all, and they seems correct now.